generate/testdata/snapshots/TestGenerateWithConfig-OptionalGeneric-testdata-queries-generated.go

```diff
@@ -0,0 +1,153 @@
+// Code generated by github.com/Khan/genqlient, DO NOT EDIT.
+
+package queries
+
+import (
+	"context"
+
+	"github.com/Khan/genqlient/graphql"
+	"github.com/Khan/genqlient/internal/testutil"
+)
+
+// ListInputQueryResponse is returned by ListInputQuery on success.
+type ListInputQueryResponse struct {
+	// user looks up a user by some stuff.
+	//
+	// See UserQueryInput for what stuff is supported.
+	// If query is null, returns the current user.
+	User testutil.Option[ListInputQueryUser] `json:"user"`
+}
+
+// GetUser returns ListInputQueryResponse.User, and is useful for accessing the field via an interface.
+func (v *ListInputQueryResponse) GetUser() testutil.Option[ListInputQueryUser] { return v.User }
+
+// ListInputQueryUser includes the requested fields of the GraphQL type User.
+// The GraphQL type's documentation follows.
+//
+// A User is a user!
+type ListInputQueryUser struct {
+	// id is the user's ID.
+	//
+	// It is stable, unique, and opaque, like all good IDs.
+	Id string `json:"id"`
+}
+
+// GetId returns ListInputQueryUser.Id, and is useful for accessing the field via an interface.
+func (v *ListInputQueryUser) GetId() string { return v.Id }
+
+// QueryWithSlicesResponse is returned by QueryWithSlices on success.
+type QueryWithSlicesResponse struct {
+	// user looks up a user by some stuff.
+	//
+	// See UserQueryInput for what stuff is supported.
+	// If query is null, returns the current user.
+	User testutil.Option[QueryWithSlicesUser] `json:"user"`
+}
+
+// GetUser returns QueryWithSlicesResponse.User, and is useful for accessing the field via an interface.
+func (v *QueryWithSlicesResponse) GetUser() testutil.Option[QueryWithSlicesUser] { return v.User }
+
+// QueryWithSlicesUser includes the requested fields of the GraphQL type User.
+// The GraphQL type's documentation follows.
+//
+// A User is a user!
+type QueryWithSlicesUser struct {
+	Emails                []string                  `json:"emails"`
+	EmailsOrNull          []string                  `json:"emailsOrNull"`
+	EmailsWithNulls       []testutil.Option[string] `json:"emailsWithNulls"`
+	EmailsWithNullsOrNull []testutil.Option[string] `json:"emailsWithNullsOrNull"`
```

Comment:
hmm, I wonder if this (which I believe corresponds to a GraphQL `[String]`, i.e. optional list of optional strings) should be `testutil.Option[[]testutil.Option[string]]`. We don't do that for pointers because it's annoying in practice, but if you're using an `Option` type it might be convenient and make it clearer for users to distinguish between `[]` and `null` using `Option`. What do you think?

Reply:
Generally speaking I prefer to use an empty array instead of option where I can because empty arrays have nice ergonomics, however you may want this in a situation where you're trying to set something to an empty array and it gets interpreted as a missing value. However I've never come across such a situation personally.